app/src/lib/stores/app-store.ts

```diff
@@ -1764,12 +1765,12 @@ export class AppStore extends TypedBaseStore<IAppState> {
   public async _loadStatus(
     repository: Repository,
     clearPartialState: boolean = false
-  ): Promise<boolean> {
+  ): Promise<IStatusResult | null> {
```

Comment:
i'm 👍 with this change, but we should also update this logic to do avoid implicit object boolean-ness: https://github.com/desktop/desktop/blob/e63ab5b33d6012d51a3e98b63563acdbcc59e3cb/app/src/lib/stores/app-store.ts#L2459